mk/distribution.mk

```diff
@@ -107,6 +107,21 @@ endif
 build/distributions/out: $(patsubst %,build/distributions/out/$(DISTRIBUTION_TARGET_NAME)-%.tar.gz,$(ENABLED_DIST_NAMES))
 	cd $@; sha256sum *.tar.gz > $(DISTRIBUTION_TARGET_NAME).sha256
 
+.PHONY: build/info/distribution/repo
+build/info/distribution/repo:
+	@echo $(PULP_PACKAGE_TYPE)-binaries-$(PULP_DIST_VERSION)
+
+.PHONY: build/info/distribution/security/pkg
+build/info/distribution/security/pkg:
+	@echo $(PULP_PACKAGE_TYPE)-seccurity-assets
+
+# Create a target to generate distribution package versions of form "<tag> / <tag>-<shortSHA> / <shortSHA>" for uploading to artifact registry
+# Needed for publishing security packages that don't leverage Kong/release-scripts
+# The "make build/info/version" generates version that is not consistent with  https://github.com/Kong/release-scripts/blob/main/cloudsmith/entrypoint.sh#L103
+.PHONY: build/info/distribution/pkg/version
+build/info/distribution/pkg/version:
+	@echo $(PACKAGE_VERSION)
```

Comment:
# Problem

- Security assets published to cloudsmith don't leverage [Kong/release-scripts](https://github.com/Kong/release-scripts/blob/main/cloudsmith/entrypoint.sh#L103)

- The `make build/info/version` produces a version that is [DIFFERENT](https://github.com/kumahq/kuma/actions/runs/9164789416/job/25197385486#step:6:8) from the version for binary packages which is constructed using the file name metadata at: [Kong/release-scripts](https://github.com/Kong/release-scripts/blob/main/cloudsmith/entrypoint.sh#L103)
![Screenshot 2024-05-20 at 5 31 20 PM](https://github.com/kumahq/kuma/assets/29549979/3630a51a-dea8-4520-9005-abf0844469f4)

# Solution

This `build/info/distribution/pkg/version` make target is created to produce a version similar to the binary-artifacts at [package_version](https://github.com/kumahq/kuma/pull/10215/files#diff-2a4ffb1dd3758151910ec681d78b7432a7fbd2b207b67975fc4c445cfba285d7R110-R128) for scenarios where artifacts uploaded to cloudsmith **DON'T** leverage `Kong/release-scripts`

Reply:
Do we agree that there are 2 cases:

- preview is in the name --> version is whatever is after the `-preview.v`
- else use as is.

Feels like we can do without playing in `version.sh` etc and just have this in the github action.

Reply:
There seems to be 3 cases:

- preview is in the name -> version is whatever is <after> the `-preview.v` where <after> is 9 digit SHORT SHA
- semver tag -> `<tag>` value
- release branch -> `<prev x.y.(z+1)>-<SHORT SHA>`

Reply:
@curiositycasualty do you know why this weirdness? Why can't we use the package version as indicated?

Reply:
It was `release-scripts` in the past that was responsible for syphoning out the short-sha from versions supplied to it that looked like `0.0.0-preview.vc1465c79a`-- but I'm having a hard time remembering why I wrote it that way... I may have had aspirations of being able to sort the repo by version, or may have thought Cloudsmith wouldn't like some of the characters in the `0.0.0-preview.v` format. But both those assumptions appear to be false; As demonstrated by the 2 styles of versions visible in the [package group view](https://cloudsmith.io/~kong/repos/kuma-binaries-preview/packages/?q=name%3A%27%5Ekuma-seccurity-assets%24%27) Cloudsmith doesn't care. And the preview repo is already pretty unsortable with either style (other than by date).

Aesthetically, I like `c1465c79a`/`2.7.3-c1465c79a` more, but this isn't my decision to make really.

I see 4 potential ways forward:
1. We decide to just be OKAY with security assets having "0.0.0-preview.v" and kuma tarballs having "<semver>-<sha>" or
2. I modify `release-script` back to use the "0.0.0-preview.v" format of version for preview packages and Sai removes the changes to `version.sh` from this PR or
3. We keep Sai's changes here and adopt the shorter "<semver>-<sha>" (no changes made to release-script) or
4. We modify `release.sh`/`make` to produce a separate version exclusively for use when uploading to Cloudsmith (of the format "<semver>-<sha>")

Where number 3 is the easiest.

Reply:
Where/how the semver-sha version is constructed in release-script: https://github.com/Kong/release-scripts/blame/main/cloudsmith/entrypoint.sh#L616-L634